test-case-reporting/types/test-case-reporting.d.ts

```diff
@@ -60,4 +60,17 @@ declare module 'test-case-reporting' {
     timestamp: Date,
     durationMs: number,
   }
+
+  export interface TravisReport {
+    successCount: number,
+    failedCount: number,
+    skippedCount: number,
+    results: Array<Result>,
+  }
+
+  export interface Result {
+    name: string,
```

Comment:
`name` is not a good label for this, since tests don't have a name, they have a description and a suite. The same is true for the label on line 52.

I'm thinking that, for example, the following block:
```ts
describe('Array', function () {
  describe('#indexOf()', function () {
    it('should return -1 when the value is not present', function () {
      assert.equal([1, 2, 3].indexOf(4), -1);
    });
  });
});
```
should be identified by the following string:
`'Array/#indexOf()/should return -1 when the value is not present'`
or maybe
`'Array/#indexOf():should return -1 when the value is not present'`
We could call this string a description, an identifier, a suite_and_description...?

Reply:
I would be in favor of either `' / '` or `' | '` as the delimiter (with spaces on either side). I'd lean toward naming the interface `TestResult`. Internally, mocha contexts offer a `name` field and `fullName()` method, so I think this nomenclature is fine. Besides, with regards to these interfaces, the goal here is to define what the JSON reporter actually uses, which isn't up to us. We'll take that data and massage it into our preferred format in the report handler.

Reply:
`> Internally, mocha contexts offer a name field and fullName() method, so I think this nomenclature is fine.`
Understood.
`> the goal here is to define what the JSON reporter actually uses, which isn't up to us.`
I've added a sample of the JSON reporter output to the PR description. Are the types good, or should they stick more closely to the JSON structure? (e.g. by have a string array for the suites).

Reply:
The point here is that these interfaces will describe the JSON reported by Travis, so that you have a type definition for the data incoming to the API endpoint. So yes, any fields defined in the interface must match the JSON format exactly. It looks like a lot of these fields need to change to match the data here

Reply:
Okay, I'll get to work on that.

Reply:
Done.